--- upstream/core/marginal_covariance_cholesky.h  (blob bff245bd0069, closest ancestor)
+++ orbslam3/core/marginal_covariance_cholesky.h
@@ -39,14 +39,13 @@
 #include <tr1/unordered_map>
 #endif
 
-#include "g2o_core_api.h"
 
 namespace g2o {
 
   /**
    * \brief computing the marginal covariance given a cholesky factor (lower triangle of the factor)
    */
-  class G2O_CORE_API MarginalCovarianceCholesky {
+  class  MarginalCovarianceCholesky {
     protected:
       /**
        * hash struct for storing the matrix elements needed to compute the covariance
