--- upstream/core/optimizable_graph.cpp  (blob e6b5071d2749, closest ancestor)
+++ orbslam3/core/optimizable_graph.cpp
@@ -41,10 +41,10 @@
 #include "cache.h"
 #include "robust_kernel.h"
 
-#include "g2o/stuff/macros.h"
-#include "g2o/stuff/color_macros.h"
-#include "g2o/stuff/string_tools.h"
-#include "g2o/stuff/misc.h"
+#include "../stuff/macros.h"
+#include "../stuff/color_macros.h"
+#include "../stuff/string_tools.h"
+#include "../stuff/misc.h"
 
 namespace g2o {
 
