--- upstream/core/robust_kernel_factory.h  (blob 41182fc81f34, closest ancestor)
+++ orbslam3/core/robust_kernel_factory.h
@@ -27,7 +27,7 @@
 #ifndef G2O_ROBUST_KERNEL_FACTORY_H
 #define G2O_ROBUST_KERNEL_FACTORY_H
 
-#include "g2o_core_api.h"
+#include "../stuff/misc.h"
 
 #include <string>
 #include <map>
@@ -41,7 +41,7 @@
     /**
    * \brief Abstract interface for allocating a robust kernel
    */
-  class G2O_CORE_API AbstractRobustKernelCreator
+  class  AbstractRobustKernelCreator
   {
     public:
       /**
@@ -64,7 +64,7 @@
   /**
    * \brief create robust kernels based on their human readable name
    */
-  class G2O_CORE_API RobustKernelFactory
+  class  RobustKernelFactory
   {
     public:
 
