--- upstream/core/hyper_graph_action.h  (blob fda5e65bd81b, closest ancestor)
+++ orbslam3/core/hyper_graph_action.h
@@ -28,7 +28,7 @@
 #define G2O_HYPER_GRAPH_ACTION_H
 
 #include "hyper_graph.h"
-#include "g2o/stuff/property.h"
+#include "../stuff/property.h"
 
 #include <typeinfo>
 #include <iosfwd>
@@ -36,7 +36,6 @@
 #include <string>
 #include <iostream>
 
-#include "g2o_core_api.h"
 
 // define to get verbose output
 //#define G2O_DEBUG_ACTIONLIB
@@ -46,14 +45,14 @@
   /**
    * \brief Abstract action that operates on an entire graph
    */
-  class G2O_CORE_API HyperGraphAction {
-    public:
-      class G2O_CORE_API Parameters {
+  class  HyperGraphAction {
+    public:
+      class  Parameters {
         public:
           virtual ~Parameters();
       };
 
-      class G2O_CORE_API ParametersIteration : public Parameters {
+      class  ParametersIteration : public Parameters {
         public:
           explicit ParametersIteration(int iter);
           int iteration;
@@ -70,9 +69,9 @@
   /**
    * \brief Abstract action that operates on a graph entity
    */
-  class G2O_CORE_API HyperGraphElementAction{
-    public:
-      struct G2O_CORE_API Parameters{
+  class  HyperGraphElementAction{
+    public:
+      struct  Parameters{
         virtual ~Parameters();
       };
       typedef std::map<std::string, HyperGraphElementAction*> ActionMap;
@@ -109,7 +108,7 @@
    * collection of actions calls contains homogeneous actions operating on different types
    * all collected actions have the same name and should have the same functionality
    */
-  class G2O_CORE_API HyperGraphElementActionCollection: public HyperGraphElementAction{
+  class  HyperGraphElementActionCollection: public HyperGraphElementAction{
     public:
       //! constructor. name_ is the name of the action e.g.draw).
       HyperGraphElementActionCollection(const std::string& name_);
@@ -134,7 +133,7 @@
    * library of actions, indexed by the action name;
    * one can use ti to register a collection of actions
    */
-  class G2O_CORE_API HyperGraphActionLibrary{
+  class  HyperGraphActionLibrary{
     public:
       //! return the single instance of the HyperGraphActionLibrary
       static HyperGraphActionLibrary* instance();
@@ -159,14 +158,14 @@
   /**
    * apply an action to all the elements of the graph.
    */
-  void G2O_CORE_API applyAction(HyperGraph* graph, HyperGraphElementAction* action, HyperGraphElementAction::Parameters* parameters=0, const std::string& typeName="");
+  void  applyAction(HyperGraph* graph, HyperGraphElementAction* action, HyperGraphElementAction::Parameters* parameters=0, const std::string& typeName="");
 
   /**
    * brief write into gnuplot
    */
-  class G2O_CORE_API WriteGnuplotAction: public HyperGraphElementAction{
-    public:
-      struct G2O_CORE_API Parameters: public HyperGraphElementAction::Parameters{
+  class  WriteGnuplotAction: public HyperGraphElementAction{
+    public:
+      struct  Parameters: public HyperGraphElementAction::Parameters{
         std::ostream* os;
       };
       WriteGnuplotAction(const std::string& typeName_);
@@ -176,9 +175,9 @@
    * \brief draw actions
    */
 
-  class G2O_CORE_API DrawAction : public HyperGraphElementAction{
+  class  DrawAction : public HyperGraphElementAction{
   public:
-    class G2O_CORE_API Parameters: public HyperGraphElementAction::Parameters,  public PropertyMap{
+    class  Parameters: public HyperGraphElementAction::Parameters,  public PropertyMap{
     public:
       Parameters();
     };
