--- upstream/core/hyper_graph.h  (blob 6f5815453493, closest ancestor)
+++ orbslam3/core/hyper_graph.h
@@ -41,7 +41,6 @@
 #include <tr1/unordered_map>
 #endif
 
-#include "g2o_core_api.h"
 
 /** @addtogroup graph */
 //@{
@@ -56,14 +55,14 @@
      The vertices are uniquely identified by an int id, while the edges are
      identfied by their pointers. 
    */
-  class G2O_CORE_API HyperGraph
+  class  HyperGraph
   {
     public:
 
       /**
        * \brief enum of all the types we have in our graphs
        */
-      enum G2O_CORE_API HyperGraphElementType {
+      enum  HyperGraphElementType {
         HGET_VERTEX,
         HGET_EDGE,
         HGET_PARAMETER,
@@ -74,13 +73,13 @@
 
       typedef std::bitset<HyperGraph::HGET_NUM_ELEMS> GraphElemBitset;
 
-      class G2O_CORE_API Vertex;
-      class G2O_CORE_API Edge;
+      class  Vertex;
+      class  Edge;
       
       /**
        * base hyper graph element, specialized in vertex and edge
        */
-      struct G2O_CORE_API HyperGraphElement {
+      struct  HyperGraphElement {
         virtual ~HyperGraphElement() {}
         /**
          * returns the type of the graph element, see HyperGraphElementType
@@ -95,7 +94,7 @@
       typedef std::vector<Vertex*>                      VertexContainer;
 
       //! abstract Vertex, your types must derive from that one
-      class G2O_CORE_API Vertex : public HyperGraphElement {
+      class  Vertex : public HyperGraphElement {
         public:
           //! creates a vertex having an ID specified by the argument
           explicit Vertex(int id=-1);
@@ -117,7 +116,7 @@
        * Abstract Edge class. Your nice edge classes should inherit from that one.
        * An hyper-edge has pointers to the vertices it connects and stores them in a vector.
        */
-      class G2O_CORE_API Edge : public HyperGraphElement {
+      class  Edge : public HyperGraphElement {
         public:
           //! creates and empty edge with no vertices
           explicit Edge(int id = -1);
