--- upstream/core/jacobian_workspace.h  (blob a66029a6a3fe, closest ancestor)
+++ orbslam3/core/jacobian_workspace.h
@@ -33,7 +33,6 @@
 #include <vector>
 #include <cassert>
 
-#include "g2o_core_api.h"
 #include "hyper_graph.h"
 
 namespace g2o {
@@ -48,7 +47,7 @@
    * Before calling linearizeOplus on an edge, the workspace needs to be allocated
    * by calling allocate().
    */
-  class G2O_CORE_API JacobianWorkspace
+  class  JacobianWorkspace
   {
     public:
       typedef std::vector<Eigen::VectorXd, Eigen::aligned_allocator<Eigen::VectorXd> >      WorkspaceVector;
