--- upstream/stuff/timeutil.h  (blob e622a751b8d1, closest ancestor)
+++ orbslam3/stuff/timeutil.h
@@ -35,7 +35,6 @@
 
 #include <string>
 
-#include "g2o_stuff_api.h"
 
 /** @addtogroup utils **/
 // @{
@@ -83,7 +82,7 @@
   long tv_sec;
   long tv_usec;
 } timeval;
-G2O_STUFF_API int gettimeofday(struct timeval *tv, struct timezone *tz);
+ int gettimeofday(struct timeval *tv, struct timezone *tz);
 #endif
 
 /**
@@ -104,7 +103,7 @@
  * On Linux we call clock_gettime() on other systems we currently
  * call get_time().
  */
-G2O_STUFF_API double get_monotonic_time();
+ double get_monotonic_time();
 
 /**
  * \brief Class to measure the time spent in a scope
@@ -112,7 +111,7 @@
  * To use this class, e.g. to measure the time spent in a function,
  * just create and instance at the beginning of the function.
  */
-class G2O_STUFF_API ScopeTime {
+class  ScopeTime {
   public: 
     ScopeTime(const char* title);
     ~ScopeTime();
