--- upstream/stuff/property.h  (blob 33162ec7647e, closest ancestor)
+++ orbslam3/stuff/property.h
@@ -32,11 +32,10 @@
 #include <sstream>
 
 #include "string_tools.h"
-#include "g2o_stuff_api.h"
 
 namespace g2o {
 
-  class G2O_STUFF_API BaseProperty {
+  class  BaseProperty {
     public:
       BaseProperty(const std::string name_);
       virtual ~BaseProperty();
@@ -73,7 +72,7 @@
   /**
    * \brief a collection of properties mapping from name to the property itself
    */
-  class G2O_STUFF_API PropertyMap : protected std::map<std::string, BaseProperty*>
+  class  PropertyMap : protected std::map<std::string, BaseProperty*>
   {
     public:
       typedef std::map<std::string, BaseProperty*>        BaseClass;
