--- upstream/DBoW2/FORB.cpp  (blob ede5d93e80c0, closest ancestor)
+++ orbslam3/DBoW2/FORB.cpp
@@ -5,13 +5,15 @@
  * Description: functions for ORB descriptors
  * License: see the LICENSE.txt file
  *
+ * Distance function has been modified 
+ *
  */
+
  
 #include <vector>
 #include <string>
 #include <sstream>
-#include <stdint.h>
-#include <limits.h>
+#include <stdint-gcc.h>
 
 #include "FORB.h"
 
@@ -20,6 +22,8 @@
 namespace DBoW2 {
 
 // --------------------------------------------------------------------------
+
+const int FORB::L=32;
 
 void FORB::meanValue(const std::vector<FORB::pDescriptor> &descriptors, 
   FORB::TDescriptor &mean)
@@ -74,44 +78,26 @@
 
 // --------------------------------------------------------------------------
   
-double FORB::distance(const FORB::TDescriptor &a, 
+int FORB::distance(const FORB::TDescriptor &a,
   const FORB::TDescriptor &b)
 {
-  // Bit count function got from:
-  // http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetKernighan
-  // This implementation assumes that a.cols (CV_8U) % sizeof(uint64_t) == 0
-  
-  const uint64_t *pa, *pb;
-  pa = a.ptr<uint64_t>(); // a & b are actually CV_8U
-  pb = b.ptr<uint64_t>(); 
-  
-  uint64_t v, ret = 0;
-  for(size_t i = 0; i < a.cols / sizeof(uint64_t); ++i, ++pa, ++pb)
+  // Bit set count operation from
+  // http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel
+
+  const int *pa = a.ptr<int32_t>();
+  const int *pb = b.ptr<int32_t>();
+
+  int dist=0;
+
+  for(int i=0; i<8; i++, pa++, pb++)
   {
-    v = *pa ^ *pb;
-    v = v - ((v >> 1) & (uint64_t)~(uint64_t)0/3);
-    v = (v & (uint64_t)~(uint64_t)0/15*3) + ((v >> 2) & 
-      (uint64_t)~(uint64_t)0/15*3);
-    v = (v + (v >> 4)) & (uint64_t)~(uint64_t)0/255*15;
-    ret += (uint64_t)(v * ((uint64_t)~(uint64_t)0/255)) >> 
-      (sizeof(uint64_t) - 1) * CHAR_BIT;
+      unsigned  int v = *pa ^ *pb;
+      v = v - ((v >> 1) & 0x55555555);
+      v = (v & 0x33333333) + ((v >> 2) & 0x33333333);
+      dist += (((v + (v >> 4)) & 0xF0F0F0F) * 0x1010101) >> 24;
   }
-  
-  return ret;
-  
-  // // If uint64_t is not defined in your system, you can try this 
-  // // portable approach (requires DUtils from DLib)
-  // const unsigned char *pa, *pb;
-  // pa = a.ptr<unsigned char>();
-  // pb = b.ptr<unsigned char>();
-  // 
-  // int ret = 0;
-  // for(int i = 0; i < a.cols; ++i, ++pa, ++pb)
-  // {
-  //   ret += DUtils::LUT::ones8bits[ *pa ^ *pb ];
-  // }
-  //  
-  // return ret;
+
+  return dist;
 }
 
 // --------------------------------------------------------------------------
@@ -185,24 +171,17 @@
 
 // --------------------------------------------------------------------------
 
-void FORB::toMat32F(const cv::Mat &descriptors, cv::Mat &mat)
-{
-  descriptors.convertTo(mat, CV_32F);
-}
-
-// --------------------------------------------------------------------------
-
 void FORB::toMat8U(const std::vector<TDescriptor> &descriptors, 
   cv::Mat &mat)
 {
-  mat.create(descriptors.size(), FORB::L, CV_8U);
+  mat.create(descriptors.size(), 32, CV_8U);
   
   unsigned char *p = mat.ptr<unsigned char>();
   
-  for(size_t i = 0; i < descriptors.size(); ++i, p += FORB::L)
+  for(size_t i = 0; i < descriptors.size(); ++i, p += 32)
   {
     const unsigned char *d = descriptors[i].ptr<unsigned char>();
-    std::copy(d, d + FORB::L, p);
+    std::copy(d, d+32, p);
   }
   
 }
@@ -211,3 +190,4 @@
 
 } // namespace DBoW2
 
+
