--- upstream/DBoW2/FeatureVector.h  (blob 08a91def7cfa, closest ancestor)
+++ orbslam3/DBoW2/FeatureVector.h
@@ -15,12 +15,22 @@
 #include <vector>
 #include <iostream>
 
+#include <boost/serialization/serialization.hpp>
+#include <boost/serialization/map.hpp>
+
 namespace DBoW2 {
 
 /// Vector of nodes with indexes of local features
 class FeatureVector: 
   public std::map<NodeId, std::vector<unsigned int> >
 {
+    friend class boost::serialization::access;
+    template<class Archive>
+    void serialize(Archive& ar, const int version)
+    {
+        ar & boost::serialization::base_object<std::map<NodeId, std::vector<unsigned int> > >(*this);
+    }
+
 public:
 
   /**
